peps/pep-0729.rst

```diff
@@ -0,0 +1,414 @@
+  checkers and does not change the overall language, would also by default
+  be decided upon by the Steering Council. However, such PEPs could be
+  delegated to the Typing Council for a decision (like any other PEP delegation).
+  Other examples of PEPs that could potentially be delegated include
+  :pep:`647`, :pep:`655`, :pep:`673`, and :pep:`675`.
+- Adding a smaller feature, such as :data:`typing.Never` as an alias for
+  :data:`typing.NoReturn`, would be done by means of a PR to the spec and
+  conformance test suite. The Typing
+  Council would then decide whether or not to merge the PR. They may ask for the
+  feature to be specified and discussed in a PEP if they feel that is warranted.
+- If there is confusion about the interpretation of some part of the spec, like
+  happened recently with `partial stubs in PEP
+  561 <https://discuss.python.org/t/pep-561-clarification-regarding-n/32875/27>`_,
+  somebody would make a PR to the typing specification to clarify the
+  spec, and then the Typing Council would decide on the spec change.
+
+The runtime :mod:`typing` module will continue to be maintained by the
+CPython core developer team. However, any changes to the runtime module that
+affect type checker behavior should be made in conjunction with a change
+to the specification (see below) and should be approved by the Typing Council.
+For example, in Python 3.11 the core developers added the new function
+:func:`typing.assert_type`. If the Typing Council had been in place, this
+change would require a matching change to the specification and approval
+by the Typing Council. On the other hand, Python 3.11 also added the
+:func:`typing.get_overloads` introspection helper. As this function does not
+affect type checker behavior, it would not require approval by the Typing
+Council.
+
+Relationship with type checkers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Typing Council has no direct authority over type checkers; it cannot
+force them to implement particular features or make behavior changes. Type
+checkers are incentivized to follow the specification set out by the Council
+because it allows them to take advantage of shared resources, such as
+libraries that expose typing information that follows the specification,
+the stub files in typeshed, the ``typing`` standard library module, and
+user documentation that covers the standard type system.
+Type checkers are free to extend the type system or deviate from the
+specification, but they should document such differences clearly.
+
+The fact that type checkers need to implement any decisions made by the
+Typing Council acts as a useful brake on the Council, ensuring that its
+decisions are conservative and well-considered. Individual type checkers
+remain free to innovate as they see fit, and successful innovations can
+be incorporated into the standard type system.
+
+Projects
+--------
+
+Here are some efforts a Typing Council would be responsible for.
+
+Conformance test suite
```

Comment:
This feels like a Sisyphean task. Aren’t we over-promising a bit?

Reply:
I think there's value in this even if it what ends up happening is that we rarely add to the test suite except when a PEP comes along and wants to add something to the type system. At the Typing Summit last year there was quite a bit of support for the idea of a conformance test suite.

Reply:
Okay, no worries. As long as we can convince people to update the test suite for new PEPs. (Which may be a problem until there's at least one complete implementation -- you can't really write tests for a feature that's not yet implemented.)